### PR TITLE
Merge from 2.2.0 final

### PR DESCRIPTION
**Pull request by JimmyShi22**

### CODE REVIEW

File: libsync/TreeTopology.cpp [hunk 1]
```diff
@@ -157,65 +142,71 @@ ssize_t TreeTopology::getParentNodeIndex(ssize_t const& _nodeIndex)
...
         {
             _selectedNodeList->push_back(selectedNode);
-            TREE_LOG(DEBUG) << LOG_DESC("selectParentNodes") << LOG_KV("parentIndex", parentIndex)
-                            << LOG_KV("selectedNode", selectedNode.abridged())
-                            << LOG_KV("idx", m_consIndex) << LOG_KV("endIndex", m_endIndex)
-                            << LOG_KV("consIndex", m_consIndex);
             break;
         }
-        if (parentIndex <= m_startIndex)
+        if (parentIndex <= 0)
```
Comment (fisco-dev) [inline libsync/TreeTopology.cpp]: Codacy Issue found: [Opposite inner 'if' condition leads to a dead code block.](https://app.codacy.com/app/fisco/FISCO-BCOS/pullRequest?prid=4712887)

**Comment by fisco-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/fisco/FISCO-BCOS/pullRequest?prid=4712887&bid=15741145)

**Comment by fisco-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/fisco/FISCO-BCOS/pullRequest?prid=4712887&bid=15741145)

File: libsync/TreeTopology.cpp (hunk 1 above)
Comment (fisco-dev) [inline libsync/TreeTopology.cpp]: Codacy Issue found: [Opposite inner 'if' condition leads to a dead code block.](https://app.codacy.com/app/fisco/FISCO-BCOS/pullRequest?prid=4712887)

**Comment by fisco-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/fisco/FISCO-BCOS/pullRequest?prid=4712887&bid=15741145)

File: libsync/TreeTopology.cpp (hunk 1 above)
Comment (fisco-dev) [inline libsync/TreeTopology.cpp]: Codacy Issue found: [Opposite inner 'if' condition leads to a dead code block.](https://app.codacy.com/app/fisco/FISCO-BCOS/pullRequest?prid=4712887)

**Comment by JimmyShi22** [comment]
Out of lines limitation. Force merge

**Pull request merged**